### PR TITLE
[dv/rom_ctrl] Add FSM exclusions

### PR DESCRIPTION
**Pull request by cindychip**
During the V2 rom_ctrl review, we noticed there are unreachable and
uncovered FSM states that need to be excluded.

Signed-off-by: Cindy Chen <chencindy@opentitan.org>

### CODE REVIEW

File: hw/ip/rom_ctrl/dv/cov/rom_ctrl_cov_unr_excl.el [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// Copyright lowRISC contributors.
```
Comment (cindychip) [inline hw/ip/rom_ctrl/dv/cov/rom_ctrl_cov_unr_excl.el:1]: @rswarbrick @prajwalaputtappa could you check if these two files correctly captured what you mentioned in the review meeting?

Comment (cindychip) [reply hw/ip/rom_ctrl/dv/cov/rom_ctrl_cov_unr_excl.el:1]: Also with UNR (VCS auto-generated), there are two uncovered FSM transactions:
image
I did not exclude them yet in manual exclusion, could you confirm if they are truly unreachable.

Comment (rswarbrick) [reply hw/ip/rom_ctrl/dv/cov/rom_ctrl_cov_unr_excl.el:1]: Thanks @cindychip!

Both of those transitions should be unreachable. The FSM, excluding the `Invalid` terminal state looks like this:
image

I'd be really interested to understand why the UNR tool understands that other transitions are impossible but not these ones, but I guess sometimes things will remain a mystery :-p

File: hw/ip/rom_ctrl/dv/cov/rom_ctrl_cov_v2_excl.el [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Manually exclude invalid state related FSM until V2S milestone is reached.
```
Comment (rswarbrick) [inline hw/ip/rom_ctrl/dv/cov/rom_ctrl_cov_v2_excl.el:5]: Could you add a TODO here (which we'll find with grep!) to remind us to remove this file once the V2S tests are up and running?

**Review by rswarbrick — APPROVED** [review]

**Review by prajwalaputtappa — APPROVED** [review]

File: hw/ip/rom_ctrl/dv/cov/rom_ctrl_cov_unr_excl.el (hunk 1 above)
Comment (cindychip) [reply hw/ip/rom_ctrl/dv/cov/rom_ctrl_cov_unr_excl.el:1]: Thanks for confirming! I will add that to another manually excluded UNR file.

File: hw/ip/rom_ctrl/dv/cov/rom_ctrl_cov_v2_excl.el (hunk 2 above)
Comment (cindychip) [reply hw/ip/rom_ctrl/dv/cov/rom_ctrl_cov_v2_excl.el:5]: Good point, thanks for pointing that out.

File: hw/ip/rom_ctrl/dv/cov/rom_ctrl_cov_excl.el [hunk 3]
```diff
@@ -0,0 +1,20 @@
+// Copyright lowRISC contributors.
```
Comment (cindychip) [inline hw/ip/rom_ctrl/dv/cov/rom_ctrl_cov_excl.el:1]: @rswarbrick @prajwalaputtappa  the latest force push added this exclusion file to capture the unreachable states that are not captured by unr.

Comment (cindychip) [reply hw/ip/rom_ctrl/dv/cov/rom_ctrl_cov_excl.el:1]: plz take a look again and I will merge it afterwards ;)

**Review by rswarbrick — APPROVED** [review]
Looks great. Thanks, @cindychip!

**Pull request merged**